web_programming/get_links_of_url.py

```diff
@@ -0,0 +1,60 @@
+import re
+import urllib.parse
+
+import requests
+from bs4 import BeautifulSoup
+
+
+def is_valid_url(url):
+    """
+    Function Valid the url
+    """
+
+    regex = re.compile(
+        r"^(?:http|ftp)s?://"
+        r"(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)"
+        r"+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|"
+        r"localhost|"
+        r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"
+        r"(?::\d+)?"
+        r"(?:/?|[/?]\S+)$",
+        re.IGNORECASE,
+    )
+    return re.match(regex, url) is not None
+
+
+def get_links(url):
+    """
+    Function get all links in website
+    """
+
+    soup = BeautifulSoup(requests.get(url).text, "html.parser")
+    data = [link.get("href") for link in soup.find_all("a")]
+    links = []
+    for link in data:
+        if not is_valid_url(link):
+            links.append(urllib.parse.urljoin(url + "/", link))
+        else:
+            links.append(link)
+    return links
+
+
+def test_get_links(url: str = "https://www.github.com") -> None:
+    """
+    A doctest for get_links function
+    >>> test_get_links()
+    """
+
+    assert len(get_links(url)) == 144
+    for url in get_links(url):
+        assert is_valid_url(url) is True
```

Comment:
That is an incorrect way to write `doctest`.
Please refer: https://docs.python.org/3/library/doctest.html
You write the test and the solution in the docstring part of a function like so:

```python
def function(a, b):
    """
    >>> function(1, 2)
    3
    """
    return a + b
```